### PR TITLE
Differential testing analytical engine: Cleanup 7715 fix with additional reviewer comments handled

### LINKED ISSUE #7715: Dump contract bytecodes and store, as text, from state file
Dump contract state - bytecodes and store - as text from a state file.  Deterministically, in the sense that contracts are dumped in the same order, k/v pairs in the same order, etc.

This will enable comparing contract state from a mono-service state file with one from a modularized service state file, where the actual representation at the platform level is different.

### PR DESCRIPTION
**Pull request by david-bakin-sl**
**Description**:

Cleanup from PR #7729, leftover reviewer comments.

Biggest single change is removing broken logging overrides.  Another useful change is fixing up the sorting of contract ids so it is really centralized in one place and applied throughout without having to take care of it.

**Related issue(s)**:

Fixes #7715

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-node/cli-clients/src/main/java/com/hedera/services/cli/signedstate/DumpContractBytecodesSubcommand.java [hunk 1]
```diff
@@ -215,15 +217,11 @@ void appendFormattedContractLines(@NonNull final StringBuilder sb, @NonNull fina
     void appendFormattedContractLine(@NonNull final StringBuilder sb, @NonNull final Contract contract) {
         sb.append(hexer.formatHex(contract.bytecode()));
 
-        final var ids = contract.ids();
-        if (withIds == DumpStateCommand.WithIds.YES && !ids.isEmpty()) {
-            // Output canonical id - we choose the minimum id (so it is deterministic)
-            final var sortedIds = ids.stream().sorted().toList();
+        if (withIds == DumpStateCommand.WithIds.YES && !contract.ids().isEmpty()) {
```
Comment (Nana-EC) [inline hedera-node/cli-clients/src/main/java/com/hedera/services/cli/signedstate/DumpContractBytecodesSubcommand.java:220]: Q: Would the `ids` assignment have been useful to make the call once?
What about?
```suggestion
        final var ids = contract.ids();
        if (withIds == DumpStateCommand.WithIds.YES && !ids.isEmpty()) {
```

Then further down
```
sb.append(ids.stream()...)
```

**Review by Nana-EC — APPROVED** [review]
LG w 1 suggestion

File: hedera-node/cli-clients/src/main/java/com/hedera/services/cli/signedstate/DumpContractBytecodesSubcommand.java (hunk 1 above)
Comment (david-bakin-sl) [reply hedera-node/cli-clients/src/main/java/com/hedera/services/cli/signedstate/DumpContractBytecodesSubcommand.java:220]: `ids()` is a simple getter from a record; per Java convention it doesn't have the `get` prefix that signifies a getter, but that's what it is.

Comment (david-bakin-sl) [reply hedera-node/cli-clients/src/main/java/com/hedera/services/cli/signedstate/DumpContractBytecodesSubcommand.java:220]: (resolving this conversatino for merge, hope you agree Nana)

**Pull request merged**